### PR TITLE
Replaced BouncyCastle lib with System.Security.Cryptography.Algorithms

### PR DESCRIPTION
**Pull request by budgetpreneur**
fix: Replaced BouncyCastle lib with System.Security.Cryptography.Algorithms

Replaced BouncyCastle lib with System.Security.Cryptography.Algorithms

fix: Added try/catch for publish operation to catch exceptions

Added try/catch for publish operation to catch exceptions

### CODE REVIEW

File: src/Api/PubnubApi/Security/PubnubCryptoBase.cs [hunk 1]
```diff
@@ -163,24 +156,11 @@ public string PubnubAccessManagerSign(string key, string data)
             byte[] keyByte = encoding.GetBytes(secret);
             byte[] messageBytes = encoding.GetBytes(message);
 
-#if NET35
             using (var hmacsha256 = new HMACSHA256(keyByte))
             {
                 byte[] hashmessage = hmacsha256.ComputeHash(messageBytes);
                 return Convert.ToBase64String(hashmessage).Replace('+', '-').Replace('/', '_');
             }
```
Comment (crimsonred) [inline src/Api/PubnubApi/Security/PubnubCryptoBase.cs:163]: Can you please confirm if this code works with UWP?

Comment (budgetpreneur) [reply src/Api/PubnubApi/Security/PubnubCryptoBase.cs:163]: Yes. It works on UWP as well. 
Tested and here is the log of request and response with signature
DateTime 11/18/2021 12:00:39, JSON= {"message":"Success","payload":{"level":"channel","subscribe_key":"sub-c-c9710928-1b7a-11e3-a0c8-02ee2ddab7fe","ttl":1440,"channels":{"my_channel_pandu":{"r":1,"w":1,"m":0,"d":0,"g":0,"u":0,"j":0}}},"service":"Access Manager","status":200} for request=http://ps.pndsn.com/v2/auth/grant/sub-key/sub-c-c9710928-1b7a-11e3-a0c8-02ee2ddab7fe?channel=my_channel_pandu&d=0&g=0&j=0&m=0&pnsdk=NETFX_CORECSharp5.4.0.0&r=1&requestid=1e2c1a26-85a0-442f-81f9-719773773a22&timestamp=1637217039&ttl=1440&u=0&uuid=winphoneuuid&w=1&signature=v2.exZfJvN_40a3zop-NNJ1CugHLgpXRHoKTAPYTfO-XYE

**Review by crimsonred — APPROVED** [review]

**Review by crimsonred — DISMISSED** [review]

File: src/Api/PubnubApi/EndPoint/PubSub/PublishOperation.cs [hunk 2]
```diff
@@ -205,67 +205,80 @@ internal void Publish(string channel, object message, bool storeInHistory, int t
...
+                        ProcessResponseCallbacks(result, requestState);
                     }
                 }
-                else
+
+                CleanUp();
+            }
+            catch (Exception ex)
+            {
+                PNStatusCategory category = PNStatusCategoryHelper.GetPNStatusCategory(400, ex.ToString());
```
Comment (crimsonred) [inline src/Api/PubnubApi/EndPoint/PubSub/PublishOperation.cs:274]: An Exception should not have the status code of 400

File: src/Api/PubnubApi/EndPoint/PubSub/PublishOperation.cs [hunk 3]
```diff
@@ -205,67 +205,80 @@ internal void Publish(string channel, object message, bool storeInHistory, int t
...
-                    ProcessResponseCallbacks(result, requestState);
+                    requestState.PubnubCallback.OnResponse(default(PNPublishResult), status);
                 }
-            }
 
-            CleanUp();
+            }
         }
 
         internal async Task<PNResult<PNPublishResult>> Publish(string channel, object message, bool storeInHistory, int ttl, Dictionary<string, object> metaData, Dictionary<string, object> externalQueryParam)
```
Comment (crimsonred) [inline src/Api/PubnubApi/EndPoint/PubSub/PublishOperation.cs:284]: Most of the code is repeated in this method

**Review by crimsonred — CHANGES_REQUESTED** [review]

File: src/Api/PubnubApi/EndPoint/PubSub/PublishOperation.cs (hunk 2 above)
Comment (budgetpreneur) [reply src/Api/PubnubApi/EndPoint/PubSub/PublishOperation.cs]: fixed it.

File: src/Api/PubnubApi/EndPoint/PubSub/PublishOperation.cs (hunk 3 above)
Comment (budgetpreneur) [reply src/Api/PubnubApi/EndPoint/PubSub/PublishOperation.cs:285]: yeah..due to async/await and callback style outputs, some code was repeated due to threading issues. In the current PR  only catch block section code was added to handle unknown exception which customer could not share. Remaining code nothing changed.

**Review by crimsonred — APPROVED** [review]

**Comment by budgetpreneur** [comment]
@client-engineering-bot release as v5.4.0.0

**Review by crimsonred — APPROVED** [review]

**Pull request merged**